server/etcdserver/bootstrap.go

```diff
@@ -328,7 +329,11 @@ func bootstrapNewClusterNoWAL(cfg config.ServerConfig, prt http.RoundTripper) (*
 	}
 	if cfg.ShouldDiscover() {
 		var str string
-		str, err = v2discovery.JoinCluster(cfg.Logger, cfg.DiscoveryURL, cfg.DiscoveryProxy, m.ID, cfg.InitialPeerURLsMap.String())
+		if cfg.EnableV2Discovery {
+			str, err = v2discovery.JoinCluster(cfg.Logger, cfg.DiscoveryURL, cfg.DiscoveryProxy, m.ID, cfg.InitialPeerURLsMap.String())
```

Comment:
Will DiscoveryProxy not be supported in V3?

Reply:
Yes, it's a good point. It will not be supported in V3 discovery. Actually I do not find a solution how to support the proxy with v3 client. 

We don't support proxy in etcdctl, so I think we can just keep it (v3 discovery) consistent with etcdctl.  

What do you think? @serathius @ptabor

Reply:
Let's add comments to both `discovery-v2` and `discovery-proxy` fields with information about deprecation in v3.7

Reply:
Thanks @serathius  for the confirmation. Just added comments.